### PR TITLE
Raven db 7205 - fix failing test

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Sparrow/AsyncManualResetEvent.cs [hunk 1]
```diff
@@ -67,19 +62,6 @@ public Task<bool> WaitAsync()
             [Pure]
             public async Task<bool> WaitAsync(TimeSpan timeout)
             {
-                Debug.Assert(timeout != TimeSpan.MaxValue);
```
Comment (fitzchak) [inline src/Sparrow/AsyncManualResetEvent.cs:70]: @ppekrol I removed it, as we need to support waiting on this value from PostponedNotificationsSender.Infinity. Please review, as I'm not sure why you added it.

File: src/Raven.Server/NotificationCenter/BackgroundWork/PostponedNotificationsSender.cs [hunk 2]
```diff
@@ -12,7 +12,7 @@ namespace Raven.Server.NotificationCenter.BackgroundWork
 {
     public class PostponedNotificationsSender : BackgroundWorkBase
     {
-        private static readonly TimeSpan Infinity = TimeSpan.FromMilliseconds(-1);
+        private static readonly TimeSpan Infinity = TimeSpan.MaxValue;
```
Comment (arekpalinski) [inline src/Raven.Server/NotificationCenter/BackgroundWork/PostponedNotificationsSender.cs]: Where do we have in `TimeoutManager` support for the notion that `TimeSpan.MaxValue` means infinite wait? 

Maybe we should get rig of infinite time span defined here and use `Timeout.InfiniteTimeSpan` which should be supported by `TimeoutManager`?

Comment (fitzchak) [reply src/Raven.Server/NotificationCenter/BackgroundWork/PostponedNotificationsSender.cs]: See the DoWork method on this class.

Comment (arekpalinski) [reply src/Raven.Server/NotificationCenter/BackgroundWork/PostponedNotificationsSender.cs]: Yes I know it's used there. I meant to remove `PostponedNotificationsSender.Infinity` and use `Timeout.InfiniteTimeSpan` in `DoWork`. 

That would require `TimeoutManager` to properly deal with `Timeout.InfiniteTimeSpan` - no need to create a timer then just return `TaskCompletionSource<object>` instance which will never have `TrySetResult` called.

File: src/Sparrow/AsyncManualResetEvent.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Sparrow/AsyncManualResetEvent.cs:70]: It was added since passing such value to `Task.Delay` resulted in throwing an exception

Comment (fitzchak) [reply src/Sparrow/AsyncManualResetEvent.cs:70]: OK. So not it does not throw as we will change it to unit.MaxValue - 1 when we create the timer.

File: src/Raven.Server/NotificationCenter/BackgroundWork/PostponedNotificationsSender.cs (hunk 2 above)
Comment (fitzchak) [reply src/Raven.Server/NotificationCenter/BackgroundWork/PostponedNotificationsSender.cs]: Done, thanks for the suggestion.

**Pull request merged**